sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/validation/ZoneValidator.java

```diff
@@ -20,52 +20,44 @@
     public static Collection<ConfigValidationError> validate(
             Optional<ServiceSpec> oldConfig,
             ServiceSpec newConfig,
-            String podType,
-            String taskName) {
+            String podType) {
         if (!oldConfig.isPresent()) {
             return Collections.emptyList();
         }
 
-        Optional<TaskSpec> oldTask = TaskUtils.getTaskSpec(oldConfig.get(), podType, taskName);
-        if (!oldTask.isPresent()) {
+        Optional<PodSpec> oldPod = getPodSpec(oldConfig.get(), podType);
+        if (!oldPod.isPresent()) {
             // Maybe the pod or task was renamed? Lets avoid enforcing whether those are rename- able and assume it's OK
             return Collections.emptyList();
         }
 
-        Optional<TaskSpec> newTask = TaskUtils.getTaskSpec(newConfig, podType, taskName);
-        if (!newTask.isPresent()) {
-            throw new IllegalArgumentException(String.format("Unable to find requested pod=%s, task=%s in config: %s",
-                    podType, taskName, newConfig));
+        Optional<PodSpec> newPod = getPodSpec(newConfig, podType);
+        if (!newPod.isPresent()) {
+            throw new IllegalArgumentException(String.format("Unable to find requested pod=%s, in config: %s",
+                    podType, newConfig));
         }
 
-        String oldEnv = oldTask.get()
-                .getCommand().get()
-                .getEnvironment()
-                .get(EnvConstants.PLACEMENT_REFERENCED_ZONE_ENV);
-        String newEnv = newTask.get()
-                .getCommand().get()
-                .getEnvironment()
-                .get(EnvConstants.PLACEMENT_REFERENCED_ZONE_ENV);
+        boolean oldReferencesZones = PlacementUtils.placementRuleReferencesZone(oldPod.get());
+        boolean newReferencesZones = PlacementUtils.placementRuleReferencesZone(newPod.get());
 
-        return validateTransition(oldEnv, newEnv, podType, taskName);
-    }
-
-    static List<ConfigValidationError> validateTransition(
-            String oldEnv, String newEnv, String podType, String taskName) {
-        boolean oldZones = Boolean.valueOf(oldEnv);
-        boolean newZones = Boolean.valueOf(newEnv);
+        Optional<PlacementRule> oldRule = oldPod.get().getPlacementRule();
+        Optional<PlacementRule> newRule = newPod.get().getPlacementRule();
 
         ConfigValidationError error = ConfigValidationError.transitionError(
```

Comment:
You could construct the error inside the conditional. Not worth redoing CI over, of course.